### ISSUE TITLE
Vendor label remains stale after initial classification

### ISSUE THREAD

**Issue by Rameen-Mahmood** [nyu-mlab/iot-inspector-client#250]
### Expected behavior

The classifier should refresh the vendor label as new device metadata becomes available, improving prediction accuracy over time.

### Actual behavior

The vendor label is set on the first classification request and never updates. Devices remain mislabeled if the initial prediction is wrong, and new features are ignored.

This is a reclassification gap: predictions run once at device creation instead of improving as more data arrives. Incremental reclassification should be triggered when new metadata is detected.

**Comment by AndrewQuijano** [comment]
A time to live of 3 minutes will be added to the cache. Let me know if this should work as a solution for you.

```python
@st.cache_data(show_spinner=False, ttl=180)
```

**Comment by AndrewQuijano** [comment]
Ah, so as it stands, your API gets called on the Device List page, which gets refreshed every second. I can look into how to separate this logic from running every second. I actually prefer your idea of event-driven only.

But I assume this means we'd need to cache the inputs for each device then right? Maybe we can check the old input, see oh, something changed, let's call the API again? As of today, the API outputs are stored in the config file. While I doubt the OUI will change, maybe we can track the remote hosts? Especially since it seems IoT Inspector will forget stuff after 60 seconds?

**Comment by AndrewQuijano** [comment]
Since the SQL Query for checking hostnames already captures hostnames seen over time, with caching, this issue should be fixed. If the input changes, then the API should be called again based on how the code works

<img width="883" height="789" alt="Image" src="https://github.com/user-attachments/assets/51504465-aa4a-4c83-a79d-becd2130acd0" />
